pkg/virt-controller/watch/vmi.go

```diff
@@ -1226,6 +1230,20 @@ func (c *VMIController) sync(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod,
 			return &syncErrorImpl{fmt.Errorf(failedToRenderLaunchManifestErrFormat, err), FailedCreatePodReason}
 		}
 
+		if c.clusterConfig.PersistentIPsEnabled() {
```

Comment:
Can this code extracted to a method and be moved under virt-controller/network?

Reply:
well yes and no
it has the following parts:
1. check the FG - need config
2. get owner VM ownerRef (need VMIController + vmi), if nil - create VMI one
3. run CreateIPAMClaims - need ipamClaimsManager
4. if err, return wrapped error 

according the reviews so far 1 and 2 were asked to run outside of CreateIPAMClaims
if we wrap it in a function:
it means they will run inside and will either require VMIController (unless we make it a VMIController method - which might be the nicest if you are fine with it)
or that it will still calculate all of those parameters outside, and pass them - in this case we did nothing but it

Reply:
addressed this by adding a function to VMIController
because config and another function is needed
hope you will like it

Reply:
`CreateNewPodIPAMClaims` is covered by a tests,
The only thing left here beside is the condition about the FG, and it is in `sync` which is a private / generic function.
In order to cover it, we need something higher level, and i don't think it worth it only for checking the FG.
It will also require unit tests in the `watch` package that are related to ipam lib, which is less nice imho.

If you still think it is required, we can consider for example adding the FG as parameter to `CreateNewPodIPAMClaims` which reduce a little bit the logic, but imho the logic here is minimal